### PR TITLE
Apollo sitesearch

### PR DESCRIPTION
**Pull request by steve-fischer-200**
I think i have covered all the bases.  Probably there will be a typo someplace, with all the field names

### CODE REVIEW

File: CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/ApolloCommentUpdaterCli.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package org.eupathdb.sitesearch.data.comments;
+
+public class ApolloCommentUpdaterCli extends CommentUpdaterCli {
+
+  public static void main(String[] args) throws Exception {
+    UserCommentUpdaterCli cli = new UserCommentUpdaterCli();
```
Comment (ryanrdoherty) [inline CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/ApolloCommentUpdaterCli.java:6]: You want ApolloCommentUpdaterCli here, right?

File: CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/CommentUpdater.java [hunk 2]
```diff
@@ -46,9 +46,14 @@
...
  *
  * It is not intended to support large cardinality updates.
+ * 
+ * NOTE: this was updated to also deal with comments coming from Apollo.  Thus the
+ * abstraction.
  *
  * @author Steve
  */
+
+// for apollo, make this class into AbstractCommentUpdater
```
Comment (ryanrdoherty) [inline CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/CommentUpdater.java:56]: This comment seems out of date.

File: CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/UserCommentUpdaterCli.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package org.eupathdb.sitesearch.data.comments;
+
+public class UserCommentUpdaterCli extends CommentUpdaterCli {
+  
+  public static void main(String[] args) throws Exception {
+    UserCommentUpdaterCli cli = new UserCommentUpdaterCli();
+    cli.init(new UserCommentSolrDocumentFields(), new UserCommentUpdaterSql());
```
Comment (ryanrdoherty) [inline CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/UserCommentUpdaterCli.java:7]: Why not pass these to the abstract class constructor rather than init()?  Looks like the class is in an invalid state before init() is called.

File: CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/ApolloCommentUpdaterCli.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    cli.init(new ApolloCommentSolrDocumentFields(), new ApolloCommentUpdaterSql());
```
Comment (ryanrdoherty) [inline CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/ApolloCommentUpdaterCli.java:7]: Why not pass these to the abstract class constructor rather than init()?  Looks like the class is in an invalid state before init() is called.

File: CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/ApolloCommentUpdaterSql.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package org.eupathdb.sitesearch.data.comments;
+
+public class ApolloCommentUpdaterSql implements CommentUpdaterSql
+{
+  @Override
+  public String getGeneIdWithCommentIdSql(String schema) {
+    // TODO Auto-generated method stub
```
Comment (ryanrdoherty) [inline CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/ApolloCommentUpdaterSql.java:7]: Needs code

File: CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/ApolloCommentUpdaterSql.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    return null;
+  }
+
+  @Override
+  public String getSortedCommentsSql(String schema) {
+    // TODO Auto-generated method stub
```
Comment (ryanrdoherty) [inline CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/ApolloCommentUpdaterSql.java:13]: Needs code

File: CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/CommentUpdater.java [hunk 7]
```diff
@@ -58,15 +63,23 @@
...
     final DatabaseInstance commentDb,
-    final String           commentSchema
+    final String           commentSchema,
+    final CommentSolrDocumentFields docFields,
+    final CommentUpdaterSql updaterSql
   ) {
     _solrUrl = solrUrl;
     _commentDb = commentDb;
-    _commentSchema = commentSchema;
+    _commentSchema = commentSchema + (commentSchema.endsWith(".") ? "" : ".");
```
Comment (ryanrdoherty) [inline CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/CommentUpdater.java:80]: Use DBPlatform.normalizeSchema() here to add the . and check a few other things.

File: CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/CommentUpdaterCli.java [hunk 8]
```diff
@@ -6,22 +6,23 @@
...
-  private static final String
-    ERR_BAD_ENV = "Comment updater requires the following environment variables:\n"
-      + "    " + ENV_DB_CONNECT + ": User database connection string\n"
-      + "    " + ENV_DB_USER    + ": User database credentials username\n"
-      + "    " + ENV_DB_PASS    + ": User database credentials password\n"
-      + "    " + ENV_DB_SCHEMA  + ": User database comment schema\n"
-      + "    " + ENV_SOLR_URL   + ": Solr URL";
+  private String getBadEnvMsg() {
+    return "Comment updater requires the following environment variables:\n"
+      + "    " + getEnvDbConnect() + ": User database connection string\n"
```
Comment (ryanrdoherty) [inline CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/CommentUpdaterCli.java:20]: These strings still specify "user" even though could be "user" or "app" now.

File: CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/CommentUpdaterCli.java [hunk 9]
```diff
@@ -32,33 +33,35 @@
     String getCommentSchema() { return getThird(); }
   }
 
-  public static void main(String[] args) throws Exception {
+  void init(CommentSolrDocumentFields docFields, CommentUpdaterSql updaterSql) throws Exception {
```
Comment (ryanrdoherty) [inline CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/CommentUpdaterCli.java:36]: Looks like this is doing a lot more than initialization.  Rename to "performUpdate()" or similar.

File: CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/CommentUpdaterCli.java [hunk 10]
```diff
@@ -32,33 +33,35 @@
...
   }
 
-  private static Config parseEnv() {
+  private Config parseEnv() {
     final var env = System.getenv();
 
-    if (!(env.containsKey(ENV_DB_CONNECT) && env.containsKey(ENV_DB_PASS)
-        && env.containsKey(ENV_DB_SCHEMA) && env.containsKey(ENV_DB_USER)
-        && env.containsKey(ENV_SOLR_URL))
+    if (!(env.containsKey(getEnvDbConnect()) && env.containsKey(getEnvDbPass())
```
Comment (ryanrdoherty) [inline CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/CommentUpdaterCli.java:49]: How is this related to the "requiredFields" method above?  Maybe could have a utility that knows how to check for a required set of keys in a map rather than this ugly if statement and use in both places.

File: CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/UserCommentUpdaterSql.java [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+        + "UNION "
+        + "SELECT stable_id "
+        + "FROM " + schema + "commentstableid "
+        + "WHERE comment_id = ?";
+  }
+
+  @Override
+  public String getSortedCommentsSql(String schema) {
+    return "SELECT source_id, comment_target_type as record_type, c.comment_id"
+    + " FROM apidb.textsearchablecomment tsc,"
```
Comment (ryanrdoherty) [inline CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/UserCommentUpdaterSql.java:19]: consistent whitespace with SQL in getGeneIdWithCommentIdSql()

**Review by ryanrdoherty — CHANGES_REQUESTED** [review]

File: CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/ApolloCommentUpdaterSql.java (hunk 6 above)
Comment (steve-fischer-200) [reply CommentUpdater/src/main/java/org/eupathdb/sitesearch/data/comments/ApolloCommentUpdaterSql.java:13]: will be supplied by DD team
